### PR TITLE
[skip circleci] Add modern issue templates

### LINKED ISSUE #388: Improve Github Issue Template
There is some stuff that is unnecessary (such as `haskell-language-server --probe-tools` since it is included in the standard verbosity), also we are using an old ISSUE_TEMPLATE format, according to github

issue_template

### LINKED ISSUE #621: Issue template doesn't work (make easier generate bug report info with no executable in path)
> Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`:

I don't have either of these on my PATH. I have no idea where they lie because VS Code installed HLS for me. Perhaps VS Code should have a "generate HLS bug report info" command we could point people at?

### PR DESCRIPTION
**Pull request by jneira**
* Closes #388
* Closes #621
* Make progress towards #2387

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2394"></a>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+about: Create a report to help us improve
+title: ''
+labels: 'status: needs triage, type: bug'
+assignees: ''
+
+---
+
+<!--
+If you encounter a bug or you have a support question, please try to fill out some of the information below.
+Please take a look to the [troubleshooting guide](https://haskell-language-server.readthedocs.io/en/latest/troubleshooting.html) before filling this new one.
```
Comment (fendor) [inline .github/ISSUE_TEMPLATE/bug_report.md:12]:
```suggestion
Please take a look at the [troubleshooting guide](https://haskell-language-server.readthedocs.io/en/latest/troubleshooting.html) before filing a new issue.
```

**Review by fendor — APPROVED** [review]
Awesome, thank you! LGTM

File: .github/ISSUE_TEMPLATE/blank-issue.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+name: Blank issue
+about: An empty template for other type of issues
+title: ''
+labels: 'status: needs triage'
+assignees: ''
+
+---
+
+<!-- Note the project has a discussions section which can be used for ask questions, make general proposals or discussions, etc -->
```
Comment (michaelpj) [inline .github/ISSUE_TEMPLATE/blank-issue.md:10]:
```suggestion
<!-- Note the project has a discussions section which can be used for to questions, make general proposals or discussions, etc -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -1,24 +1,26 @@
...
+name: Bug report
+about: Create a report to help us improve
+title: ''
+labels: 'status: needs triage, type: bug'
+assignees: ''
+
+---
+
 <!--
 If you encounter a bug or you have a support question, please try to fill out some of the information below.
```
Comment (michaelpj) [inline .github/ISSUE_TEMPLATE/bug_report.md:11]: For some reason it won't let me make a suggestion here, but here's what I suggest:
```
Before opening an issue, please take a look at the [troubleshooting guide](https://haskell-language-server.readthedocs.io/en/latest/troubleshooting.html). This explains some common issues and will also help you to find the information that the issue template asks for.

When filing an issue, please fill out as much of the information below as you can. This helps us to debug your issue, but is not required!
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -1,24 +1,26 @@
...
-Generally speaking, the information below is meant to help debugging issues but is no prerequisite for opening an issue.
+Please take a look at the [troubleshooting guide](https://haskell-language-server.readthedocs.io/en/latest/troubleshooting.html) before filing a new issue.
+The information below is meant to help debugging issues but is no prerequisite for opening an issue.
 -->
 
 ### Your environment
-Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`:
```
Comment (michaelpj) [inline .github/ISSUE_TEMPLATE/bug_report.md:7]: This is still useful, no?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -30,8 +32,8 @@ Contents of `hie.yaml`:
 <!-- Tell us what happens instead. -->
 
 ### Include debug information
-Execute in the root of your project the command `haskell-language-server --debug .` and paste the logs here:
-
+Execute in the root of your project the command `haskell-language-server-wrapper --debug .` and paste the logs here:
+(if you are using the vscode extension check the executable location [here](https://github.com/haskell/vscode-haskell#downloaded-binaries))
```
Comment (michaelpj) [inline .github/ISSUE_TEMPLATE/bug_report.md:36]: I wonder if we should just write more about this on the troubleshooting page, and then here we could say "If you don't know where to find the `haskell-language-server` exectuable, consult the troubleshooting page"

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -42,7 +44,7 @@ Debug output:
 ```
 </details>
 
-Paste the logs from the lsp-client, e.g. for [VS Code](https://github.com/haskell/vscode-haskell#troubleshooting)
+Paste the logs from the lsp-client, you can check instructions about for VS Code [here](https://github.com/haskell/vscode-haskell#troubleshooting)
```
Comment (michaelpj) [inline .github/ISSUE_TEMPLATE/bug_report.md:47]: Again, I think we should refer people to the relevant section of the troubleshooting page.

**Review by michaelpj — APPROVED** [review]
If you want to get really fancy, you could use the new issue forms: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

My main thought is that I think as much as possible we should enhance the troubleshooting page and direct people there, so the information is centralised in one place. I'm hoping to make a PR for some improvements there in a bit!

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 4 above)
Comment (fendor) [reply .github/ISSUE_TEMPLATE/bug_report.md:7]: `haskell-language-server --debug .` also prints the tools, so I think this is indeed superfluous.

File: .github/ISSUE_TEMPLATE/blank-issue.md (hunk 2 above)
Comment (jneira) [reply .github/ISSUE_TEMPLATE/blank-issue.md:10]: i guess the correction would be

```suggestion
<!-- Note the project has a discussions section which can be used to ask questions, make general proposals or discussions, etc -->
```

**Comment by jneira** [comment]
> If you want to get really fancy, you could use the new issue forms: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

wow didnt know about that feature, looks really nice. I think i will give a try

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (jneira) [reply .github/ISSUE_TEMPLATE/bug_report.md]: done!

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (jneira) [reply .github/ISSUE_TEMPLATE/bug_report.md:36]: Yeah, although a direct reference to the location may be helpful too? I mean, we are linking the troubleshooting page in the beginning but in this point the focus is locate it and the table linked gives you the path

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 6 above)
Comment (jneira) [reply .github/ISSUE_TEMPLATE/bug_report.md:47]: hmm https://github.com/haskell/vscode-haskell#troubleshooting would not be the relevant page?

**Comment by jneira** [comment]
> My main thought is that I think as much as possible we should enhance the troubleshooting page and direct people there, so the information is centralised in one place. I'm hoping to make a PR for some improvements there in a bit!

what do you think revisit the issue template after improving the troubleshooting section and update the links?

**Comment by michaelpj** [comment]
> what do you think revisit the issue template after improving the troubleshooting section and update the links?

Yes, fine!

**Comment by jneira** [comment]
> wow didnt know about that feature, looks really nice. I think i will give a try

It looks nice but it will take some time to get it right so i think is better to improve content and switch to forms in other pr

**Pull request merged**